### PR TITLE
tetex: remove override libpng

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

#259879

with `libpng` it does not compile.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pbsds — DISMISSED** [review]

**Comment by pbsds** [comment]
I'm not blocking, but in #270107 we found that the override should be kept, not inlined.

**Comment by Stunkymonkey** [comment]
> I'm not blocking, but in #259879 we found that the override should be kept, not inlined.

I am reading: `If the override is still necessary then move it directly to the default.nix with a comment about why.`

**Comment by pbsds** [comment]
And in that thread I and kirilldry challenge the issue, since the by-name README states that using `libpng12` as input instead of `libpng` changes the override interface and is therefore discouraged.

File: pkgs/tools/typesetting/tex/tetex/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchurl, flex, bison, zlib, libpng, ncurses, ed, automake }:
+{ lib, stdenv, fetchurl, flex, bison, zlib, libpng12, libpng ? libpng12, ncurses, ed, automake }:
```
Comment (pbsds) [inline pkgs/tools/typesetting/tex/tetex/default.nix:1]:
```
error: builder for '/nix/store/hapr60a6qhjwd2hhf5ljz2334pmfp5ll-tetex-3.0.drv' failed with exit code 2;
       last 10 log lines:
       > writepng.c:152:45: error: invalid use of incomplete typedef 'png_info' {aka 'struct png_info_def'}
       >   152 |             pdfbuf[pdfptr++] = png_info(img)->palette[i].blue;
       >       |                                             ^~
```

The issue is explained here https://github.com/NixOS/nixpkgs/issues/259879#issuecomment-1827743352

**Pull request closed without merging**